internal/services/oracle/oracle_network_anchor_resource.go

```diff
@@ -0,0 +1,402 @@
+
+type NetworkAnchorResource struct{}
+
+type NetworkAnchorResourceModel struct {
+	Location          string            `tfschema:"location"`
+	Name              string            `tfschema:"name"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	Tags              map[string]string `tfschema:"tags"`
+	Zones             zones.Schema      `tfschema:"zones"`
+
+	ResourceAnchorId string `tfschema:"resource_anchor_id"`
+	SubnetId         string `tfschema:"subnet_id"`
+
+	OciVcnDnsLabel                     string                   `tfschema:"oci_vcn_dns_label"`
+	OciBackupCidrBlock                 string                   `tfschema:"oci_backup_cidr_block"`
+	OracleDnsForwardingEndpointEnabled bool                     `tfschema:"oracle_dns_forwarding_endpoint_enabled"`
+	OracleDnsListeningEndpointEnabled  bool                     `tfschema:"oracle_dns_listening_endpoint_enabled"`
+	OracleToAzureDnsZoneSyncEnabled    bool                     `tfschema:"oracle_to_azure_dns_zone_sync_enabled"`
+	DnsForwardingRuleUrl               string                   `tfschema:"dns_forwarding_rule_url"`
+	DnsForwardingRule                  []DnsForwardingRuleModel `tfschema:"dns_forwarding_rule"`
+	DnsForwardingEndpointIpAddress     string                   `tfschema:"dns_forwarding_endpoint_ip_address"`
+	DnsForwardingEndpointNsgRuleUrl    string                   `tfschema:"dns_forwarding_endpoint_nsg_rule_url"`
+	DnsListeningEndpointAllowedCidrs   string                   `tfschema:"dns_listening_endpoint_allowed_cidr"`
+	DnsListeningEndpointIpAddress      string                   `tfschema:"dns_listening_endpoint_ip_address"`
+	DnsListeningEndpointNsgRuleUrl     string                   `tfschema:"dns_listening_endpoint_nsg_rule_url"`
+}
+
+type DnsForwardingRuleModel struct {
+	DomainNames         string `tfschema:"domain_names"`
+	ForwardingIPAddress string `tfschema:"forwarding_ip_address"`
+}
+
+func (NetworkAnchorResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.Location(),
+
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(3, 24),
+				validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9-]+$`), "Name may include letters, numbers, or hyphens only"),
+			),
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"resource_anchor_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"oci_vcn_dns_label": {
+			Type: pluginsdk.TypeString,
+			// O+C if the value not specified, this gets set to the name of the Network Anchor
+			Optional: true,
+			Computed: true,
+			ForceNew: true,
```

Comment:
While this is supported it is somewhat unusual in the provider, can we add a `Description` explaining this?
```suggestion
			Computed: true,
			Description:  "If the `oci_vcn_dns_label` is omitted it will be set to the `name` field of the network anchor resource.",
			ForceNew: true,
```

Reply:
A description was added